chain/network/src/peer.rs

```diff
@@ -107,9 +166,13 @@ impl Peer {
         let network_client_msg = match msg {
             PeerMessage::Block(block) => {
                 // TODO: add tracking of requests here.
-                NetworkClientMessages::Block(block, peer_id, false)
+                let block_hash = b.hash();
+                self.tracker.push_recevied(block_hash);
+                NetworkClientMessages::Block(block, peer_id, self.tracker.has_request(block_hash))
             }
             PeerMessage::BlockHeaderAnnounce(header) => {
+                let block_hash = b.hash();
+                self.tracker.push_recevied(block_hash);
```

Comment:
Why only pushing header and block to tracker?

Reply:
Transaction tracking will be separate as part of the "tx routing" #1038